Fix/playground routing breadcrumb

dejongbaba opened the pull request.

MohammedMaaz commented on agenta-web/src/components/Layout/Layout.tsx:
```diff
@@ -120,6 +120,36 @@ const App: React.FC<LayoutProps> = ({children}) => {
+        return [{title: <Link href="/apps">Apps</Link>}, {title: capitalizedAppName}]
+    }
+
+    const breadCrumbItems = computePlaygroundBreadCrumbs()
```
please wrap this inside useMemo to avoid unnecessary recomputations:

`const breadCrumbItems = useMemo(computePlaygroundBreadCrumbs, [capitalizedAppName, router.pathname])`

Please include any other dependencies if I've missed

MohammedMaaz commented:
I know that it was in the description of the issue to use variantName in the path like `apps/{app_name}/playground/{variant_name}`. However, I'd suggest to go with having `variantName` as a query param and remove the new page (`apps/[app_name]/playground/[variant_name].tsx`) you created.

MohammedMaaz commented:
Can you please also check your code for types, lint and prettier issues and fix them so the failing workflows pass?

dejongbaba commented:
> Can you please also check your code for types, lint and prettier issues and fix them so the failing workflows pass?

Yes, I just noticed this morning, would cater to all the fixes.

MohammedMaaz commented:
> > Can you please also check your code for types, lint and prettier issues and fix them so the failing workflows pass?
> 
> Yes, I just noticed this morning, would cater to all the fixes.

Hi @dejongbaba 

Any update on this?

aakrem reviewed (changes requested):
Hi @dejongbaba 
I appreciate your effort on this PR!
I have these changes requests:

1. I believe our routing for variants can be optimized:
Given that variants function as standalone entities on our platform, it would make more sense to use path parameters. A structure like /playground/variants/{variantName} might offer a clearer and more direct navigation for users compared to /products?variant=XYZ.

This way, our URL structure would echo the structure of the data model from the REST API we're consuming.

2. When we hit the endpoint directly from the website we should be able to display the right tab. Right now we have an error doing that
<img width="1453" alt="Screenshot 2023-09-28 at 07 55 33" src="https://github.com/Agenta-AI/agenta/assets/6608260/e80e0fd1-dced-480b-b73f-b4f77c25dd50">

dejongbaba commented:
> Hi @dejongbaba I appreciate your effort on this PR! I have these changes requests:
> 
> 1. I believe our routing for variants can be optimized:
>    Given that variants function as standalone entities on our platform, it would make more sense to use path parameters. A structure like /playground/{variantName} might offer a clearer and more direct navigation for users compared to /playground?variant=XYZ.
> 
> This way, our URL structure would echo the structure of the data model from the REST API we're consuming.
> 
> 2. When we hit the endpoint directly from the website we should be able to display the right tab. Right now we have an error doing that
> 
> <img alt="Screenshot 2023-09-28 at 07 55 33" width="1453" src="https://user-images.githubusercontent.com/6608260/271191658-e80e0fd1-dced-480b-b73f-b4f77c25dd50.png">

Hi @aakrem  I made use of the path parameters as suggested earlier, my code was reviewed by @MohammedMaaz and he said using query parameters was more preferable, hence the change in implementation.

dejongbaba commented:
> I know that it was in the description of the issue to use variantName in the path like `apps/{app_name}/playground/{variant_name}`. However, I'd suggest to go with having `variantName` as a query param and remove the new page (`apps/[app_name]/playground/[variant_name].tsx`) you created.

Hi @aakrem here is where @MohammedMaaz mentioned I should change to query params.

aakrem commented:
Hey @dejongbaba. Understood. Sorry for the misalignments.
I will try to make sure the reviewers are also aligned on the requirements.
Do you still wanna work on the requested changes ?

dejongbaba commented:
> Hey @dejongbaba. Understood. Sorry for the misalignments. I will try to make sure the reviewers are also aligned on the requirements. Do you still wanna work on the requested changes ?

Yes sure, as far as you are aligned with the other reviewers and give the go ahead.

aakrem commented:
@dejongbaba now we are! please go ahead :)

aakrem reviewed (changes requested):
Hi @dejongbaba 
am still getting redirected http://localhost/apps/yo/playground?variant_name=v1

dejongbaba commented:
> Hi @dejongbaba am still getting redirected http://localhost/apps/yo/playground?variant_name=v1



> Hi @dejongbaba am still getting redirected http://localhost/apps/yo/playground?variant_name=v1

Everything seems to be good on here, please can you describe when you get this redirects ?

aakrem commented:
Hi @dejongbaba, I retested this and now am getting this after creating a new app from template:

https://github.com/Agenta-AI/agenta/assets/6608260/f1110de7-7bab-4701-8008-79e61b7b72d4

dejongbaba commented:
> Hi @dejongbaba, I retested this and now am getting this after creating a new app from template:
> 
>  Screen.Recording.2023-09-30.at.13.23.33.mov

dejongbaba commented:
> Hi @dejongbaba, I retested this and now am getting this after creating a new app from template:
> 
>  Screen.Recording.2023-09-30.at.13.23.33.mov

hi @aakrem please help take a look again, everything should be good now.

aakrem commented on agenta-web/src/pages/apps/[app_name]/playground/[variant_name].tsx:
It looks we are not using this component @dejongbaba ?

dejongbaba replied:
we are actually, it's how we are able to get the variant name route parameter in the playground component. it helps to update the url when we click on the variants.

aakrem commented on agenta-web/package-lock.json:
can we also remove this one as there is no new dependencies introduced ?

dejongbaba replied:
okay would remove this

aakrem replied:
Hi @dejongbaba, I saw that you completely removed the package-lock.
What was meant by removing it; removing it from the PR, not removing it completely from the codebase :) 
It means we don't need to have changes in package-lock as long as we haven't introduced new dependencies

aakrem reviewed (changes requested):
Nice, now it's working like how it should. It looks cool!!
I added small comments @dejongbaba, once resolved we can merge this one.

aakrem reviewed (changes requested):
hey @dejongbaba. I saw that you removed the package-lock .
What was meant by removing it, is removing from the PR not from from the codebase :) 
It means we don't need to have changes in the package-lock as long as we haven't introduced any new dependencies.

So put back the package-lock without adding any changes in it.

aakrem reviewed (changes requested):
You are still removing the whole file!

<img width="1477" alt="Screenshot 2023-10-05 at 14 20 22" src="https://github.com/Agenta-AI/agenta/assets/6608260/fca9c346-296b-4473-b71f-59b67900880a">

A very simple solution is to copy the content of the [package-lock](https://github.com/Agenta-AI/agenta/blob/main/agenta-web/package-lock.json) from the `main` branch and push it

dejongbaba commented:
> You are still removing the whole file!
> 
> <img alt="Screenshot 2023-10-05 at 14 20 22" width="1477" src="https://user-images.githubusercontent.com/6608260/272900084-fca9c346-296b-4473-b71f-59b67900880a.png">
> A very simple solution is to copy the content of the [package-lock](https://github.com/Agenta-AI/agenta/blob/main/agenta-web/package-lock.json) from the `main` branch and push it

okay , that's cool thanks for the suggestion

dejongbaba commented:
@aakrem Done as you suggested, how ever I don't think the package-lock affects the application that’s why I removed it initially. So one the application builds it regenerates a new package-lock.json that has all the dependencies needed for the app. If it was a scenario were I added a new package then it the updates package.json would be needed but this pr the package-lock.json isn't needed. Let me know what you think about this.

aakrem commented:
@dejongbaba let's fix this together. join us in [slack](https://join.slack.com/t/agenta-hq/shared_invite/zt-1zsafop5i-Y7~ZySbhRZvKVPV5DO_7IA)

MohammedMaaz commented:
> @aakrem Done as you suggested, how ever I don't think the package-lock affects the application that’s why I removed it initially. So once the application builds it regenerates a new package-lock.json that has all the dependencies needed for the app. If it was a scenario were I added a new package then updates in the package.json would be needed but for this pr the package-lock.json isn't needed. Let me know what you think about this.

@dejongbaba you are correct, this PR doesn't need any changes in the package-lock.json. The fact that your PR shows changes in package-lock.json is because of Next JS swc compiler behavior on different OS. Its a known issue and is in our todo to solve. For now, what you can do to not include pcakage-lock changes in your PR is to just copy the contents of package-lock.json file from the main branch to your branch and push.

MohammedMaaz commented:
@dejongbaba also please resolve the conflicts and your PR is good to be merged
